rust/src/combinators/user_defined/ffi.rs

```diff
@@ -0,0 +1,281 @@
+                        InsertDeleteDistance,
+                        ChangeOneDistance,
+                        HammingDistance,
+                        DiscreteDistance
+                    ]
+                )],
+                ()
+            )
+        }
+        TypeContents::GENERIC { .. } => {
+            let QA = M.get_atom()?;
+            fn monomorphize<QA>(M: Type) -> Fallible<AnyMetric>
+            where
+                QA: 'static + CheckNull,
+            {
+                fn monomorphize<M>() -> Fallible<AnyMetric>
+                where
+                    M: 'static + Metric + Default,
+                {
+                    Ok(AnyMetric::new(M::default()))
+                }
+                dispatch!(monomorphize, [(M, [AbsoluteDistance<QA>, L1Distance<QA>, L2Distance<QA>])], ())
+            }
+            dispatch!(monomorphize, [(QA, @numbers)], (M))
+        }
+        _ => fallible!(FFI, "unrecognized metric: {}", M.to_string()),
+    }
+}
+pub(crate) fn default_measure(M: Type) -> Fallible<AnyMeasure> {
+    let QA = M.get_atom()?;
+    fn monomorphize<QA>(M: Type) -> Fallible<AnyMeasure>
+    where
+        QA: 'static + CheckNull,
+    {
+        fn monomorphize<M>() -> Fallible<AnyMeasure>
+        where
+            M: 'static + Measure + Default,
+        {
+            Ok(AnyMeasure::new(M::default()))
+        }
+        dispatch!(monomorphize, [(M, [MaxDivergence<QA>, FixedSmoothedMaxDivergence<QA>, ZeroConcentratedDivergence<QA>])], ())
+    }
+    dispatch!(monomorphize, [(QA, @numbers)], (M))
+}
+
+#[bootstrap(
+    name = "make_default_transformation",
+    features("contrib", "honest-but-curious"),
+    arguments(
+        function(rust_type = "$domain_carrier_type(DO)"),
+        stability_map(rust_type = "$metric_distance_type(MO)"),
+        DI(rust_type = b"null", hint = "RuntimeTypeDescriptor"),
+        DO(rust_type = b"null", hint = "RuntimeTypeDescriptor"),
+        MI(rust_type = b"null", hint = "RuntimeTypeDescriptor"),
+        MO(rust_type = b"null", hint = "RuntimeTypeDescriptor"),
+    ),
+    dependencies("c_function", "c_stability_map")
+)]
+/// Construct a Transformation from user-defined callbacks.
+/// 
+/// **Supported Domains:**
+/// 
+/// * `VectorDomain<AllDomain<_>>`
+/// * `AllDomain<_>`
+/// 
+/// **Supported Metrics:**
+/// 
+/// * `SymmetricDistance`
+/// * `InsertDeleteDistance`
+/// * `ChangeOneDistance`
+/// * `HammingDistance`
+/// * `DiscreteDistance`
+/// * `AbsoluteDistance<_>`
+/// * `L1Distance<_>`
+/// * `L2Distance<_>`
+/// 
+/// # Arguments
+/// * `function` - A function mapping data from `DI` to `DO`.
+/// * `stability_map` - A function mapping distances from `MI` to `MO`.
+/// * `DI` - Input Domain. See Supported Domains
+/// * `DO` - Output Domain. See Supported Domains
+/// * `MI` - Input Metric. See Supported Metrics
+/// * `MO` - Output Metric. See Supported Metrics
+#[no_mangle]
+pub extern "C" fn opendp_combinators__make_default_transformation(
```

Comment:
It feels a little awkward to call these `make_default_xxx`, as the default part seems incidental. The emphasis here is on the user part, so how about `make_user_xxx`? If we later have a way to pass in domains etc, we could have that be an optional argument, or make a different entry point.

Reply:
I was trying to guard the `make_transformation` namespace. This reflects the limitation that the domains you choose must impl `Default`. Keep in mind, in the future, when we change the arguments to accept custom domains, the arguments will get reordered. Does this context influence your naming suggestion?